addons/point_of_sale/static/src/js/models.js

```diff
@@ -580,6 +634,180 @@ export class PosGlobalState extends PosModel {
             i += 1;
         } while (partners.length);
     }
+    setLoadingOrderState(bool) {
+        this.loadingOrderState = bool;
+    }
+    async _removeOrdersFromServer() {
+        const removedOrdersIds = this.db.get_ids_to_remove_from_server();
+        if (removedOrdersIds.length === 0) {
+            return;
+        }
+
+        const timeout = TIMEOUT * removedOrdersIds.length;
+        this.set_synch("connecting", removedOrdersIds.length);
+        try {
+            const removeOrdersResponseData = await this.env.services.rpc(
+                {
+                    model: "pos.order",
+                    method: "remove_from_ui",
+                    args: [removedOrdersIds],
+                },
+                {
+                    timeout: timeout,
+                    shadow: true,
+                }
+            );
+            this.set_synch("connected");
+            this._postRemoveFromServer(removedOrdersIds, removeOrdersResponseData);
+        } catch (reason) {
+            const error = reason.message;
+            if (error.code === 200) {
+                // Business Logic Error, not a connection problem
+                //if warning do not need to display traceback!!
+                if (error.data.exception_type == "warning") {
+                    delete error.data.debug;
+                }
+            }
+            // important to throw error here and let the rendering component handle the error
+            console.warn("Failed to remove orders:", removedOrdersIds);
+            throw error;
+        }
+    }
+    _postRemoveFromServer(serverIds, data) {
+        this.db.set_ids_removed_from_server(serverIds);
+    }
+    _replaceOrders(ordersToReplace, newOrdersJsons) {
+        ordersToReplace.forEach((order) => {
+            // We don't remove the validated orders because we still want to see them in the ticket screen.
+            // Orders in 'ReceiptScreen' or 'TipScreen' are validated orders.
+            if ((this.selectedOrder.uid != order.uid) && order.server_id && !order.finalized) {
```

Comment:
I see that you added a new condition, what is it used for? From my understanding the synchronization is done in the TicketScreen, so if the current order has been updated elsewhere, we don't update it  ??